src/ast/semantic_analyser.cpp

```diff
@@ -31,6 +32,24 @@ static const std::map<std::string, std::tuple<size_t, bool>>& getIntcasts() {
   return intcasts;
 }
 
+// TODO: (fbs) We should get rid of this macro at some point
+#define ERR(x, loc)                                                            \
```

Comment:
Somewhat strange pattern. One thing I've done in the past is make the macro ostream-like: https://github.com/facebookincubator/oomd/blob/master/src/oomd/Log.h#L145 . The ::Oomd::LogStream() destructor sends stuff out the appropriate sink.

This is how it's used: https://github.com/facebookincubator/oomd/blob/master/src/oomd/config/ConfigCompiler.cpp#L64

IMO this makes it a little more intuitive b/c you think of the macro itself as being an ostream object rather than the first parameter being sent to an ostream. In this case, the usage would look like:
```
ERR(loc) << "asdf" << binop;
```